File: ACT3.py
class Calculator:
    def __init__(self, passcode):
        self.__secretPasscode = passcode
        
    def unlock_secret(self, passcode):
        return passcode == self.__secretPasscode
        
    def add(self, a, b):
        return a + b
        
    def subtract(self, a, b):
        return a - b
        
    def multiply(self, a, b):
        return a * b
        
    def divide(self, a, b):
        return a / b if b != 0 else "Error: Division by zero"
        
    def power(self, a, b, passcode):
        if self.unlock_secret(passcode):
            return a ** b
        else:
            return "Access Denied. We are proceeding with basic opeartions only."
    
    
def main():
    user_passcode = input("Enter the passcode to unlock the secret operation: ")
    
    calc = Calculator(user_passcode)
    
    a = float(input("Enter first number: "))
    b = float(input("Enter second number: "))
    
    print("\n\nChoose an operation:")
    print("1 - Add")
    print("2 - Subtract")
    print("3 - Multiply")
    print("4 - Divide")
    print("5 - Power (Secret Operation)")
        
    choice = (input("Enter your choice: "))
    
    if choice in ['1', '2', '3', '4', '5']:
        
        if choice == '1':
            result = calc.add(a, b)
        elif choice == '2':
            result = calc.subtract(a, b)
        elif choice == '3':
            result = calc.multiply(a, b)
        elif choice == '4':
            result = calc.divide(a, b)
        elif choice == '5':
            passcode = input("\nEnter passcode to unlock the secret opeartion: ")
            result = calc.power(a, b, passcode)
            
        print(f"\nResult: {result}")
        
    else:
        print("Invalid choice!")
        
if __name__ == "__main__":
        main()
    
        
            
    
    
    
    
    
    
    
    
    
    

File: ACT3.java
import java.util.Scanner;
class Calculator{
    private String secretPasscode;
    
    public Calculator(String passcode){
        this.secretPasscode = passcode;
    }
    
    private boolean unlockSecret(String passcode){
        return this.secretPasscode.equals(passcode);
    }
    
    public double add(double a, double b){
        return a + b;
    }
    
    public double subtract(double a, double b){
        return a - b;
    }
    
    public double multiply(double a, double b){
        return a * b;
    }
    
    public String divide(double a, double b){
        if (b != 0){
            return String.valueOf(a / b);
        } else {
            return "Error: Division by zero";
        }
    }
    
    public String power(double a, double b, String passcode){
        if (unlockSecret(passcode)){
            return String.valueOf(Math.pow(a, b));
        } else {
            return "Access denied. We are proceeding with basic operations only!";
        }
    }
}


class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter a new passcode to unlok secret operation: ");
        String userPasscode = scanner.nextLine();
        
        Calculator calc = new Calculator(userPasscode);
        
        System.out.print("Enter first number: ");
        double a = scanner.nextDouble();
        
        System.out.print("Enter second number: ");
        double b = scanner.nextDouble();
       
        
        System.out.println("\nChoose an operation:");
        System.out.println("1- ADD");
        System.out.println("2 - SUBTRACT");
        System.out.println("3 - MULTIPLY");
        System.out.println("4 - DIVIDE");
        System.out.println("5 - POWER (Secret Operation)");
        
        System.out.print("Enter your choice: ");
        String choice = scanner.nextLine();
        
        String result;
        
        switch (choice){
            case "1":
                result = String.valueOf(calc.add(a, b));
            case "2":
                result = String.valueOf(calc.subtract(a, b));
            case "3":
                result = String.valueOf(calc.multiply(a, b));
            case "4":
                result = String.valueOf(calc.divide(a, b));
            case "5":
                scanner.nextLine();
                System.out.print("\nEnter passcode to Unlock the secret operation: ");
                String passcode = scanner.nextLine();
                result = calc.power(a, b , passcode);
                break;
            default:
                result = "Invalid choice!";
        }
        
        System.out.println("\nResult: "+ result);
        
        
    }
}
